pkg/pillar/pubsub/updaters.go

```diff
@@ -30,7 +30,8 @@ func (u *Updaters) Add(updater chan Notify, name string, instance int) {
 // Remove an updater
 func (u *Updaters) Remove(updater chan Notify) {
 	u.lock.Lock()
-	servers := make([]notifyName, len(u.servers))
+	// A new slice which points to the same underlying array
+	servers := u.servers[:0]
```

Comment:
The base array is being mangled at the same time as being iterated over.
If u.servers starts with a length of 5 and we delete (prevent from getting appended) one of the elements, the final u.servers base array will still have 5 elements (old remnant). This code will basically overwrite the first 4 elements in base array and leave the 5th element as it is. Isn't is better to initialize a new slice?

Reply:
The storage for the 5th remains in place, and its content. But it isn't part of the slice since the length of the slice is 4.

You mean allocate a new array; this is a new slice onto the same array.

Reply:
Sure, but you are not updating the variable u.servers with new slice.
u.servers is a slice with original length (say N) but the first N -1 elements updated.

Reply:
Yep. Fixed that.